Swift/Sources/StateMachine/StateMachine.swift

```diff
@@ -109,6 +130,12 @@ open class StateMachine<State: StateMachineHashable, Event: StateMachineHashable
                 if let toState: State = action.toState {
                     state = toState
                 }
+
+                if stateIdentifier != state.hashableIdentifier {
+                    onExitActions[stateIdentifier]?()
+                    onEnterActions[state.hashableIdentifier]?()
```

Comment:
Or transition side-effects happen after you enter the new state? This also seems odd.